detection_rules/rule.py

```diff
@@ -457,6 +460,20 @@ def process_note_plugins():
 
         return obj
 
+    @validates_schema
+    def validates_data(self, data, **kwargs):
+        """Validate version and revision fields not supplied for Elastic authored rules."""
```

Comment:
This is a multi-purpose method, so to generalize:

```suggestion
        """Validate fields and data for marshmallow schemas."""
        # validate version and revision fields not supplied for Elastic authored rules
```

Even better (and self-documenting) would be to create a method specific to this and call it within `validates_data`